dwave/plugins/torch/samplers/block_spin_sampler.py

```diff
@@ -0,0 +1,348 @@
+        crayon (Callable[Hashable, Hashable]): A colouring function; a function that maps a single
+            node of the `grbm` to its colour.
+        num_chains (int): Number of Markov chains to run in parallel.
+        initial_states (torch.Tensor | None): A tensor of +/-1 values of shape
+            (num_chains, grbm.n_nodes) Initial states of the Markov chains. If None, initial
+            states will be uniformly randomized with number of chains equal to ``num_chains``.
+            Defaults to None.
+        schedule (Iterable[Float]): The inverse temperature schedule.
+        proposal_acceptance_criteria (Literal["Gibbs", "Metropolis"]): The proposal acceptance
+            criterion used to accept or reject states in the Markov chain. Defaults to "Gibbs".
+        seed (Optional[int]): Random seed. Defaults to None.
+    Raises:
+        InvalidProposalAcceptanceCriteriaError: If the proposal acceptance criteria is not one of
+            "Gibbs" or "Metropolis".
+    """
+
+    def to(self, device: DeviceLikeType) -> None:
+        """Moves sampler components to the target device in-place.
+
+        If the device is "meta", then the random number generator ``BlockSpinSampler._rng`` (RNG)
+        will not be modified at all. For all other devices, all fields ``_x``, ``_zeros``,
+        ``_schedule``, ``_partition``, ``_padded_adjacencies``, and ``_padded_adjacencies_weight``
+        will be moved to the target device. Importantly, the RNG's device is relayed by the
+        following procedure:
+        1. Draw a random integer between 0 (inclusive) and 2**60 (exclusive) with the current
+           generator as a new seed ``s``.
+        2. Create a new generator on the target device.
+        3. Set the new generator's seed as ``s``.
+
+        Args:
+            device (DeviceLikeType): The target device.
+        """
+        self._x = self._x.to(device)
+        self._zeros = self._zeros.to(device)
+        self._schedule = self._schedule.to(device)
+        self._partition = self._partition.to(device)
+        self._padded_adjacencies = self._padded_adjacencies.to(device)
+        self._padded_adjacencies_weight = self._padded_adjacencies_weight.to(device)
+        if device != "meta":
+            rng = torch.Generator(device)
+            rng.manual_seed(torch.randint(0, 2**60, (1,), generator=self._rng).item())
+            self._rng = rng
+
+    def __init__(self, grbm: GRBM, crayon: Callable[[Hashable], Hashable], num_chains: int,
+                 schedule: Iterable[float],
+                 proposal_acceptance_criteria: Literal["Gibbs", "Metropolis"] = "Gibbs",
+                 initial_states: torch.Tensor | None = None,
+                 seed: int | None = None):
+        super().__init__()
+
+        if not isinstance(num_chains, int) or num_chains < 1:
+            raise NonPositiveIntegerError(
+                "Number of reads should be a positive integer."
+            )
```

Comment:
Since this is both a `TypeError` and `ValueError` check you could have `NonPositiveIntegerError` inherit from both, which would allow it to be caught by both.
```python
class NonPositiveIntegerError(TyperError, ValueError):
    """TODO"""
```

Reply:
(Removed per other comment)